x/evm/keeper/state_transition.go

```diff
@@ -64,13 +67,29 @@ func (k *Keeper) NewEVM(
 		BaseFee:     cfg.BaseFee,
 		Random:      nil, // not supported
 	}
-
 	txCtx := core.NewEVMTxContext(msg)
 	if tracer == nil {
 		tracer = k.Tracer(ctx, msg, cfg.ChainConfig)
 	}
 	vmConfig := k.VMConfig(ctx, msg, cfg, tracer)
-	return k.evmConstructor(blockCtx, txCtx, stateDB, cfg.ChainConfig, vmConfig, k.customPrecompiles)
+	rules := cfg.ChainConfig.Rules(big.NewInt(ctx.BlockHeight()), cfg.ChainConfig.MergeNetsplitBlock != nil)
+	contracts := make(map[common.Address]vm.PrecompiledContract)
+	active := make([]common.Address, 0)
+	for addr, c := range vm.DefaultPrecompiles(rules) {
+		contracts[addr] = c
+		active = append(active, addr)
+	}
+	for _, c := range customContracts {
+		addr := c.Address()
+		contracts[addr] = c
+		active = append(active, addr)
+	}
+	sort.SliceStable(active, func(i, j int) bool {
+		return bytes.Compare(active[i].Bytes(), active[j].Bytes()) < 0
+	})
```

Comment:
can we cache the sorted list in the keeper, save some cpu cycles?

Reply:
keeper use customContractsFn instead of customContracts, since need ctx and stateDB when [new](https://github.com/crypto-org-chain/cronos/blob/e267528f9c9557ef9320ad40cf8372d8ed7b01ec/x/cronos/keeper/precompiles/relayer.go#L37) contract and there's no stateDB when NewKeeper